aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/runtime/XmlUtility.java

```diff
@@ -167,6 +168,24 @@
      */
     void mapProperties(Element element, Annotation annotation, List<String> skippedFields);
 
+    /**
+     * Appends {@link DataSource} values to an {@code Element} node
+     * @param element Element to store data in
+     * @param acsListPath Path to ACS Commons List in JCR repository
+     * @param acsListResourceType Use this to set {@code sling:resourceType} of data source, other than standard
+     * @param dataSource Provided values as a {@code DataSource} annotation
+     */
+    void appendDataSource(Element element, DataSource dataSource, String acsListPath, String acsListResourceType);
+
+    /**
+     * Appends to the current {@code Element} node and returns a child {@code datasource} node
+     * @param element Element to store data in
+     * @param path Path to element
+     * @param resourceType Use this to set {@code sling:resourceType} of data source
+     * @return Appended {@code datasource} node
+     */
+    Element appendDataSource(Element element,  String path, String resourceType, Map<String, String> properties);
```

Comment:
Two concerns?

1) Do we need to add these to the public interface? Adding datasource is not an universal method, this is for the two particular cases. 
*Especially* the acs-list-bound one, which is deprecated through and through!

I myself stay opposed,
There are indeed methods in `PluginXmlUtility` that are not exposed in this interface.

2) Looks a bit  strange -- the two overloading methods, similar in meaning, yet one has a return value, the other hasn't. Is this for reason?